packages/contracts-bedrock/test/setup/Setup.sol

```diff
@@ -120,9 +121,18 @@ contract Setup {
     ///      This is a hack as we are pushing solidity to the edge.
     function setUp() public virtual {
         console.log("L1 setup start!");
-        vm.etch(address(deploy), vm.getDeployedCode("Deploy.s.sol:Deploy"));
+        if (vm.envOr("UPGRADE_TEST", false)) {
```

Comment:
It would be good to add some comments here about what is happening so that its easier to onboard future maintainers

Reply:
I see there are some comments in `Upgrade.s.sol`, those are nice